contracts/orbital-core/src/contract.rs

```diff
@@ -1,119 +1,171 @@
     Ok(Response::new())
 }
 
 #[entry_point]
 pub fn execute(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     env: Env,
     info: MessageInfo,
     msg: ExecuteMsg,
-) -> Result<Response, ContractError> {
+) -> NeutronResult<Response<NeutronMsg>> {
     match msg {
+        // admin action to manage ownership of orbital-core
         ExecuteMsg::UpdateOwnership(action) => {
-            admin_update_ownership(deps, &env.block, &info.sender, action)
+            admin::transfer_admin(deps, &env.block, &info.sender, action)
         }
+        // admin action to enable new domain for user registration
         ExecuteMsg::RegisterNewDomain {
             domain,
             account_type,
-        } => admin_register_new_domain(deps, info, domain, account_type),
-        ExecuteMsg::RegisterUser {} => register_user(deps, env, info),
+        } => admin::register_new_domain(deps, info, domain, account_type),
+        // user action to create a new user account which enables registration to domains
+        ExecuteMsg::RegisterUser {} => user::register(deps, env, info),
+        // user action to register a new domain which creates their clearing account
+        ExecuteMsg::RegisterUserDomain { domain } => {
+            user::register_new_domain(deps, env, info, domain)
+        }
     }
 }
 
-fn register_user(deps: DepsMut, _env: Env, info: MessageInfo) -> Result<Response, ContractError> {
-    // user can only register once
-    if USER_CONFIGS.has(deps.storage, info.sender.clone()) {
-        return Err(ContractError::UserAlreadyRegistered {});
+#[entry_point]
+pub fn query(deps: QueryDeps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::OrbitalDomain { domain } => to_json_binary(&query_orbital_domain(deps, domain)?),
+        QueryMsg::UserConfig { addr } => to_json_binary(&query_user_config(deps, addr)?),
+        QueryMsg::Ownership {} => to_json_binary(&query_ownership(deps)?),
+        QueryMsg::ClearingAccountAddress { addr, domain } => {
+            to_json_binary(&query_clearing_account(deps, domain, addr)?)
+        }
+        QueryMsg::UserAddresses {} => to_json_binary(&query_user_addresses(deps)?),
     }
+}
 
-    // save an empty user config
-    USER_CONFIGS.save(deps.storage, info.sender, &UserConfig::default())?;
+fn query_user_addresses(deps: QueryDeps) -> StdResult<Vec<String>> {
+    let users = USER_CONFIGS.range(deps.storage, None, None, cosmwasm_std::Order::Ascending);
 
-    Ok(Response::new().add_attribute("method", "register_user"))
-}
+    let mut user_addrs = vec![];
+    for user_entry in users {
+        user_addrs.push(user_entry?.0);
+    }
 
-fn admin_update_ownership(
-    deps: DepsMut,
-    block: &BlockInfo,
-    sender: &Addr,
-    action: Action,
-) -> Result<Response, ContractError> {
-    let resp = update_ownership(deps, block, sender, action)?;
-    Ok(Response::default().add_attributes(resp.into_attributes()))
+    Ok(user_addrs)
```

Comment:
nit: I'd probably rewrite this as something like:

```rust
    let limit = limit.unwrap_or(MAX_PAGE_LIMIT).min(MAX_PAGE_LIMIT);
    let start = start_after.map(Bound::exclusive);

    USER_CONFIGS
        .range(deps.storage, start, None, Order::Ascending)
        .take(limit as usize)
        .filter_map(Result::ok)
        .map(|config| config.id.to_string())
        .collect()
```       
or something like that (after pagination is included). I think it's more optimized than loading, creating a vec and pushing 1 by 1. But I think it looks cleaner

Reply:
sorry quoted too many lines, meant the query_user_addresses func

Reply:
removed this entirely, sorry again!